src/fixup/fixup.cpp

```diff
@@ -645,6 +664,156 @@ TaskStatus ApplyFloorsImpl(T *rc, IndexDomain domain = IndexDomain::entire) {
+
+  const bool enable_phi_enforcement = fix_pkg->Param<bool>("enable_phi_enforcement");
+
+  namespace p = fluid_prim;
+  namespace c = fluid_cons;
+  const std::vector<std::string> vars({p::bfield, c::bfield});
+  PackIndexMap imap;
+  auto pack = md->PackVariables(vars, imap);
+
+  const int pblo = imap[p::bfield].first;
+  const int cblo = imap[c::bfield].first;
+
+  const auto ib = md->GetBoundsI(IndexDomain::interior);
+  const auto jb = md->GetBoundsJ(IndexDomain::interior);
+  const auto kb = md->GetBoundsK(IndexDomain::interior);
+
+  auto geom = Geometry::GetCoordinateSystem(md);
+  auto gpkg = pm->packages.Get("geometry");
+  bool derefine_poles = gpkg->Param<bool>("derefine_poles");
+  Real h = gpkg->Param<Real>("h");
+  Real xt = gpkg->Param<Real>("xt");
+  Real alpha = gpkg->Param<Real>("alpha");
+  Real x0 = gpkg->Param<Real>("x0");
+  Real smooth = gpkg->Param<Real>("smooth");
+  auto tr = Geometry::McKinneyGammieRyan(derefine_poles, h, xt, alpha, x0, smooth);
+
+  if (enable_phi_enforcement) {
+    const Real enforced_phi = fix_pkg->Param<Real>("enforced_phi");
+    const Real enforced_phi_timescale = fix_pkg->Param<Real>("enforced_phi_timescale");
+    const Real enforced_phi_cadence = fix_pkg->Param<Real>("enforced_phi_cadence");
+    const Real enforced_phi_start_time = fix_pkg->Param<Real>("enforced_phi_start_time");
+    if (t > enforced_phi_start_time) {
+
+      // This only needs to be done at initialization
+      ParArrayND<Real> A("vector potential", pack.GetDim(5), jb.e + 2, ib.e + 2);
+      // Update B field
+      parthenon::par_for(
+          parthenon::LoopPatternMDRange(), "Phoebus::Fixup::EndOfStepModify::EvaluateQ",
+          DevExecSpace(), 0, pack.GetDim(5) - 1, jb.s, jb.e + 1, ib.s, ib.e + 1,
+          KOKKOS_LAMBDA(const int b, const int j, const int i) {
+            const auto &coords = pack.GetCoords(b);
+            const Real x1 = coords.Xf<1>(0, j, i);
+            const Real x2 = coords.Xf<2>(0, j, i);
+            const Real r = tr.bl_radius(x1);
+            const Real th = tr.bl_theta(x1, x2);
+
+            const Real x = r * sin(th);
+            const Real z = r * cos(th);
+            const Real a_hyp = 1.2;
+            const Real b_hyp = 3. * a_hyp;
+            const Real x_hyp = a_hyp * sqrt(1. + pow(z / b_hyp, 2.));
+            const Real q = (pow(x, 2) - pow(x_hyp, 2)) / pow(x_hyp, 2.);
+            if (x < x_hyp) {
+              A(b, j, i) = q;
+            }
+          });
+
+      // Recalculate dphi_dt occasionally
+      if (last_stage && t >= next_dphi_dt_update_time) {
+        next_dphi_dt_update_time += enforced_phi_cadence;
+
+        // Record mdot
+        const Real mdot_proc = History::ReduceMassAccretionRate(md);
+        const Real mdot = reduction::Sum(mdot_proc);
+
+        // Measure phi
+        const Real phi_proc = History::ReduceMagneticFluxPhi(md);
+        const Real phi = reduction::Sum(phi_proc) / std::sqrt(mdot);
```

Comment:
Note of caution here---if we have more than one `MeshData` object per MPI rank, this might hang? Probably better to use task-based reducers... or enforce somewhere that this is the only `MeshData` object on a given MPI rank for this task region

Reply:
Yeah I need to not be lazy and just use the parthenon reductions. I'll make that change...

Reply:
I did it! Though maybe I should put the `enable_phi_enforcement` check in the task list itself for brevity...